src/Runner.Worker/FileCommandManager.cs

```diff
@@ -0,0 +1,140 @@
+﻿using GitHub.Runner.Worker.Container;
+using System;
+using System.Text;
+using System.IO;
+using GitHub.Runner.Common;
+using GitHub.Runner.Sdk;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace GitHub.Runner.Worker
+{
+    [ServiceLocator(Default = typeof(FileCommandManager))]
+    public interface IFileCommandManager : IRunnerService
+    {
+        void InitializeFiles(IExecutionContext context, ContainerInfo container);
+        void TryProcessFiles(IExecutionContext context, ContainerInfo container);
+    
+    }
+
+    public sealed class FileCommandManager : RunnerService, IFileCommandManager
+    {
+        private const string _folderName = "_runner_file_commands";
+        private List<IFileCommandExtension> _commandExtensions;
+        private string _fileSuffix = String.Empty;
+        private string _fileCommandDirectory;
+        private Tracing _trace;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _trace = HostContext.GetTrace(nameof(FileCommandManager));
+
+            _fileCommandDirectory = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), _folderName);
+            if (!Directory.Exists(_fileCommandDirectory))
+            {
+                Directory.CreateDirectory(_fileCommandDirectory);
+            }
+
+            var extensionManager = hostContext.GetService<IExtensionManager>();
+            _commandExtensions = extensionManager.GetExtensions<IFileCommandExtension>() ?? new List<IFileCommandExtension>();
+            
+        }
+
+        public void InitializeFiles(IExecutionContext context, ContainerInfo container)
+        {
+            var oldSuffix = _fileSuffix;
+            _fileSuffix = Guid.NewGuid().ToString();
+            foreach (var fileCommand in _commandExtensions)
+            {
+                var oldPath = Path.Combine(_fileCommandDirectory, fileCommand.FileName + oldSuffix);
+                if (oldSuffix != String.Empty && File.Exists(oldPath))
+                {
+                    TryDeleteFile(oldPath);
+                }
+
+                var newPath = Path.Combine(_fileCommandDirectory, fileCommand.FileName + _fileSuffix);
+                TryDeleteFile(newPath);
+                File.Create(newPath).Dispose();
+
+                var pathToSet = container != null ? container.TranslateToContainerPath(newPath) : newPath; 
+                context.SetGitHubContext(fileCommand.ContextName, pathToSet);
```

Comment:
nit: would be rad if more explicitly set in ActionRunner. Hard to mentally keep track where the global context is changed. Today it's only StepsRunner/CompositeActionRunner. May be drawback of plugin abstraction.